packages/ra-core/src/controller/useListParams.ts

```diff
@@ -229,38 +240,18 @@ const useListParams = ({
     );
 
     const hideFilter = useCallback((filterName: string) => {
-        // we don't use lodash.set() for displayed filters
```

Comment:
I moved the logic to the queryReducer for easier testability, but that has no effect on the bug.